pkg/provider/storage/azure_storageaccount.go

```diff
@@ -321,11 +384,29 @@ func (az *Cloud) EnsureStorageAccount(ctx context.Context, accountOptions *Accou
 	accountOptions.Tags[consts.CreatedByTag] = "azure"
 
 	var createNewAccount bool
+	clientFactory := az.ComputeClientFactory
+	if clientFactory == nil {
+		return "", "", fmt.Errorf("clientFactory is nil")
+	}
+	var err error
+	var storageAccountClient accountclient.Interface
+	if accountOptions.SubscriptionID != "" {
+		storageAccountClient, err = clientFactory.GetAccountClientForSub(accountOptions.SubscriptionID)
+		if err != nil {
+			return "", "", err
+		}
+	} else {
+		storageAccountClient = clientFactory.GetAccountClient()
```

Comment:
I think you could always use `storageAccountClient, err = clientFactory.GetAccountClientForSub(accountOptions.SubscriptionID)` since if `accountOptions.SubscriptionID` is empty, then finally `clientFactory.GetAccountClient()` would be called, that make the logic simpler

Reply:
fixed.